### PR TITLE
perf: special-case antiquotation syntax in Pratt parser to avoid choice nodes

### PR DESCRIPTION
**Pull request by Kha**

### CODE REVIEW

File: src/Init/Lean/Parser/Term.lean [hunk 1]
```diff
@@ -98,7 +98,6 @@ withPosition $ fun pos =>
 @[builtinTermParser] def borrowed   := parser! symbol "@&" appPrec >> termParser (appPrec - 1)
 @[builtinTermParser] def quotedName := parser! nameLit
 -- NOTE: syntax quotations are defined in Init.Lean.Parser.Command
-@[builtinTermParser] def antiquot   := (mkAntiquot "term" none true : Parser)
```
Comment (leodemoura) [inline src/Init/Lean/Parser/Term.lean]: Can we also remove the following definition at `Command.lean`?
```lean
@[builtinCommandParser] def Command.antiquot := (mkAntiquot "command" none true : Parser)
```

Comment (Kha) [reply src/Init/Lean/Parser/Term.lean]: Yes, good catch! Done.

**Comment by leodemoura** [comment]
merged

**Comment by Kha** [comment]
Wow, parsing went up 22% http://speedcenter.informatik.kit.edu/lean/changes/?tre=10&rev=4279df417063a03500c7b47b81332cf74acfe698&exe=37&env=2

**Comment by leodemoura** [comment]
I pushed the following commit to (try to) address the perf issue:
https://github.com/leanprover/lean4/commit/3a9f6e41332e5d091d3ef725060d50d85ce0d0a7
It improved the runtime on my machine. Let's see on the speedcenter :)

**Comment by Kha** [comment]
Nice! Looks like it has completely negated the overhead, though we are still quite a bit slower than we were before any antiquotations.

**Comment by leodemoura** [comment]
Not sure whether this is the only change is impacting performance.
After you added antiquotations, I refactored the implementation to support user-defined syntax categories. Before this change, we had one environment extension per category. Now, we have only one, but we have this extra check.
https://github.com/leanprover/lean4/blob/master/src/Init/Lean/Parser/Parser.lean#L1818

BTW, I pushed a minor performance improvement.
https://github.com/leanprover/lean4/commit/9169a975f8f3586a2b242e719b0d76b703ea4ac6

**Comment by leodemoura** [comment]
Sorry, the link above is wrong. Here is the right one:
https://github.com/leanprover/lean4/blob/master/src/Init/Lean/Parser/Parser.lean#L1825

**Pull request closed without merging**